server/handler/ai/db.go

```diff
@@ -0,0 +1,148 @@
+		return nil, err
+	}
+	return reports, nil
+}
+
+// GenerateReport creates a new report based on the latest 7days PRs and issues data and save it in database.
+func GenerateReport(db *gorm.DB) (models.Report, error) {
+	endTime := time.Now()
+	startTime := endTime.AddDate(0, 0, -7)
+
+	// Fetch merged pull requests
+	var pullRequests []models.PullRequest
+	if err := db.Where("state = ? AND created_at BETWEEN ? AND ?", "MERGED", startTime, endTime).
+		Preload("Author").Find(&pullRequests).Error; err != nil {
+		return models.Report{}, err
+	}
+
+	// Extract required fields from pull requests
+	var formattedPullRequests []map[string]interface{}
+	for _, pr := range pullRequests {
+		var authorData map[string]interface{}
+		if pr.Author != nil {
+			authorData = map[string]interface{}{
+				"login": pr.Author.Login,
+				"name":  pr.Author.Name,
+				"bio":   pr.Author.Bio,
+			}
+		}
+		formattedPullRequests = append(formattedPullRequests, map[string]interface{}{
+			"updatedAt":    pr.UpdatedAt,
+			"repositoryID": pr.RepositoryID,
+			"title":        pr.Title,
+			"author":       authorData,
+		})
+	}
+
+	// Fetch open issues
+	var issues []models.Issue
+	if err := db.Where("state = ? AND created_at BETWEEN ? AND ?", "OPEN", startTime, endTime).
+		Preload("Author").Find(&issues).Error; err != nil {
+		return models.Report{}, err
+	}
+
+	// Extract required fields from issues
+	var formattedIssues []map[string]interface{}
+	for _, issue := range issues {
+		var authorData map[string]interface{}
+		if issue.Author != nil {
+			authorData = map[string]interface{}{
+				"login": issue.Author.Login,
+				"name":  issue.Author.Name,
+				"bio":   issue.Author.Bio,
+			}
+		}
+		formattedIssues = append(formattedIssues, map[string]interface{}{
+			"updatedAt":    issue.UpdatedAt,
+			"repositoryID": issue.RepositoryID,
+			"title":        issue.Title,
+			"author":       authorData,
+		})
+	}
+
+	// Prepare response data
+	responseData := map[string]interface{}{
+		"pullRequests": formattedPullRequests,
+		"issues":       formattedIssues,
+	}
+
+	// fmt.Printf("Choices: %+v\n", responseData)
+
+	responseJSON, err := json.Marshal(responseData)
+	if err != nil {
+		return models.Report{}, err
+	}
+
+	// Generate report using AskAssistant
+	assistantResponse, err := providers.AskMistral(reportSystemPrompt, string(responseJSON), reportOutputFormatSchema)
+	if err != nil {
+		return models.Report{}, err
+	}
+
+	// Parse assistantResponse into a valid JSON format
+	var parsedData map[string]interface{}
+	if err := json.Unmarshal([]byte(assistantResponse), &parsedData); err != nil {
+		return models.Report{}, fmt.Errorf("invalid JSON format in assistant response: %w", err)
+	}
+
+	// Convert parsedData back to a JSON string
+	formattedData, err := json.Marshal(parsedData)
+	if err != nil {
+		return models.Report{}, fmt.Errorf("failed to format assistant response as JSON: %w", err)
+	}
+
+	// Create a new report instance
```

Comment:
I think it would be a good idea to store also the input that we used to generated the report,
that could be useful to debug in case a report is not satisfactory.
at least we can have the `user prompt`

Reply:
I don't think is heavy, and I think this could be useful. Maybe you can Store just what changes on your input not the entire input, but I don't think is neccessary